yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -390,8 +390,9 @@ private[spark] class Client(
     // Upload Spark and the application JAR to the remote file system if necessary,
     // and add them as local resources to the application master.
     val fs = destDir.getFileSystem(hadoopConf)
-    val nns = YarnSparkHadoopUtil.get.getNameNodesToAccess(sparkConf) + destDir
-    YarnSparkHadoopUtil.get.obtainTokensForNamenodes(nns, hadoopConf, credentials)
+    hdfsTokenProvider(sparkConf).setNameNodesToAccess(sparkConf, Set(destDir))
```

Comment:
+1; it would be better if all interactions with token providers were done through the common interface; it seems like these HDFS-specific calls could easily be moved to the HDFS token provider.